pkg/provisioning/service.go

```diff
@@ -43,40 +48,79 @@ func NewService(
 	plService PipelineService,
 	connService ConnectorService,
 	procService ProcessorService,
+	pipelinesDir string,
 ) *Service {
 	return &Service{
 		db:               db,
 		logger:           logger.WithComponent("provisioning.Service"),
 		pipelineService:  plService,
 		connectorService: connService,
 		processorService: procService,
+		pipelinesPath:    pipelinesDir,
 	}
 }
 
-func (s *Service) ProvisionConfigFile(ctx context.Context, path string) ([]string, error) {
+func (s *Service) Init(ctx context.Context) error {
+	s.logger.Debug(ctx).Msg("initializing the provisioning service")
+
+	var multierr error
+	var files []string
+	err := filepath.WalkDir(s.pipelinesPath, func(path string, fileInfo fs.DirEntry, err error) error {
+		if strings.HasSuffix(path, ".yml") {
+			files = append(files, path)
+		}
+		return nil
+	})
+	if err != nil {
+		multierr = multierror.Append(multierr, cerrors.Errorf("could not iterate through the pipelines folder %q: %w", s.pipelinesPath, err))
+		return multierr
+	}
+
+	if len(files) == 0 {
+		s.logger.Warn(ctx).Str("pipelinesDir", s.pipelinesPath).Msg("configuration folder has no YAML files, no pipelines created")
+	}
+
+	var pipelines []string
+	for _, file := range files {
+		provPipelines, err := s.provisionConfigFile(ctx, file)
+		if err != nil {
+			multierr = multierror.Append(multierr, err)
+			continue
+		}
+		pipelines = append(pipelines, provPipelines...)
+	}
+	s.logger.Info(ctx).Int("count", len(pipelines)).Str("pipelines", strings.Join(pipelines, ", ")).Msg("pipelines successfully provisioned")
+
+	s.deleteOldPipelines(ctx, pipelines)
+
+	return multierr
+}
+
+func (s *Service) provisionConfigFile(ctx context.Context, path string) ([]string, error) {
+	var multierr error
+
 	data, err := os.ReadFile(path)
 	if err != nil {
 		return nil, cerrors.Errorf("could not read the file %q: %w", path, err)
 	}
 
 	before, err := Parse(data)
 	if err != nil {
-		return nil, err
+		return nil, cerrors.Errorf("could not parse the file %q: %w", path, err)
 	}
 
 	got := EnrichPipelinesConfig(before)
 
 	var pls []string
-	var multierr error
```

Comment:
Out of curiosity, why did you move the declaration a few lines above, when `multierr` is not used?

Reply:
Oh I was gonna add the error from `Parse(data)` to it.. so moved it at the top.. but then changed it back